packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.ts

```diff
@@ -83,13 +85,19 @@ export const useTreeViewExpansion: TreeViewPlugin<UseTreeViewExpansionSignature>
     }
   };
 
+  const isEditingEnabled = hasPlugin(instance, useTreeViewLabel);
+
   const expansionTrigger = React.useMemo(() => {
     if (params.expansionTrigger) {
       return params.expansionTrigger;
     }
 
+    if (isEditingEnabled && instance.isTreeViewEditable) {
```

Comment:
I did not think about this problem... We can't really use `hasPlugin` in a `useMemo` :/

Maybe we can just check the method existence here, the `isEditingEnabled` does not add any value right?